enterprise-suite/scripts/install-es.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import sys 
+import os
+import glob
+import shlex
+import shutil
+import subprocess
+import threading
+import tempfile
+import re
+import argparse
+from distutils.version import LooseVersion
+
+DEFAULT_TIMEOUT=3
+
+# Parsed commandline args
+args = None
+
+# Runs a given command with optional timeout.
+# Returns (stdout, returncode) tuple. If timeout
+# occured, returncode will be negative (-9 on macOS).
+def run(cmd, timeout=None, stdin=None):
+    stdout, stderr, returncode = None, None, None
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdout=subprocess.PIPE,
+                                stdin=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        timer = threading.Timer(timeout, proc.kill) if timeout != None else None 
+        if timer != None:
+            timer.start()
+        stdout, stderr = proc.communicate(input=stdin)
+        returncode = proc.returncode
+    finally:
+        if timer != None:
+            timer.cancel()
+        return stdout, returncode
+
+# Executes a command if dry_run=False,
+# prints it to stdout, handles failure status
+# codes by exiting with and error if can_fail=False.
+def execute(cmd, can_fail=False):
+    print(cmd)
+    if not args.dry_run:
+        stdout, returncode = run(cmd)
+        print(stdout)
+        if not can_fail and returncode != 0:
+            sys.exit("Command '" + cmd + "' failed:\n" + stdout)
+        return returncode
+    return 0
+
+version_re = re.compile(r'([0-9]+\.[0-9]+(\.[0-9]+)?)')
+def require_version(cmd, required_version):
+    # Use first word as a program name in messages
+    name = cmd.partition(' ')[0]
+
+    # Use 1s timeout, mainly for docker when DOCKER_HOST is unreachable
+    stdout, returncode = run(cmd, 1)
+
+    if returncode == None:
+        sys.exit("Required program '" + name + "' not found")
+    elif returncode == 0 and stdout != '':
+        match = version_re.search(stdout)
+        if match != None:
+            current = LooseVersion(match.group())
+            required = LooseVersion(required_version)
+            if current >= required:
+                return
+            else:
+                sys.exit("Installed version of '" + name + "' is too old. Found: {}, required: {}"
+                    .format(current, required)) 
+
+    # Non-critical warning
+    print("warning: unable to determine installed version of '" + name + "'")
+
+def preflight_check(creds, minikube=False):
+    # Check versions
+    require_version("docker version -f '{{.Client.Version}}'", '17.06')
```

Comment:
Can the version numbers be defined as constants?

Reply:
Sure, changed that